soc/arm/gigadevice/common/pinctrl_soc.h

```diff
@@ -106,12 +106,19 @@ typedef uint32_t pinctrl_soc_pin_t;
 #ifdef CONFIG_PINCTRL_GD32_AF
 /** Maximum 2MHz */
 #define GD32_OSPEED_2MHZ 0U
+#ifdef CONFIG_SOC_SERIES_GD32F3X0
```

Comment:
I'm more in favor to have a HAS_GPIO_V1/V2 stuff to select this options.
This will avoid big lists of CONFIG_SOC_FOO entries.

@gmarull ?

Reply:
sounds reasonable, do you know if GD has V1/V2 IP only?

Reply:
No I don't know. It is just an idea that occurred me when I was studding clocks. Until now we touched many Series and two or maybe 3 HAS_ names it is not a problem. This could help touch less code when add a newer SoC series too.

But the name not sound good, maybe need think something better. It could be good only define a "standard" way to name those things to help with other drivers.

Not a block, just an idea.